bigtop_toolchain/manifests/packages.pp

```diff
@@ -63,7 +63,10 @@
         "libffi-devel",
         "krb5-devel",
         "net-tools",
-        "perl-Digest-SHA"
+        "perl-Digest-SHA",
+        "centos-release-scl",
+        "ninja-build",
+        "glibc-devel.i686"
```

Comment:
The package listed here should work for all of RHEL, CentOS and Fedora regardless platform. It looks assuming CentOS and i686.

Reply:
`glibc-devel.i686` is just for x86 on Centos-7. 
In this conditional branch(`centos|fedora|redhat `), it doesn't work for the other Distros and the other architectures(`Arm64 `and `ppcle64`).
@kevinw66

Reply:
Thanks for the reminding, is `gcc-multilib` on Ubuntu also only for x86? @guyuqi

Reply:
Yes, on ubuntu-20.04:
`apt search gcc-multilib:`

```
gcc-multilib/focal,now 4:9.3.0-1ubuntu2 amd64 [installed]
  GNU C compiler (multilib files)

```

The other architectures packages of Arm, mips, powerpc64 : `gcc-multilib` + `arch`:
```
gcc-multilib-arm-linux-gnueabi/focal 4:9.3.0-1ubuntu2 amd64
  GNU C compiler for the armel architecture

gcc-multilib-arm-linux-gnueabihf/focal 4:9.3.0-1ubuntu2 amd64
  GNU C compiler for the armhf architecture

gcc-multilib-i686-linux-gnu/focal 4:9.3.0-1ubuntu2 amd64
  GNU C compiler for the i386 architecture

gcc-multilib-mips-linux-gnu/focal 4:9.2.1-4.1 amd64
  GNU C compiler for the mips architecture

gcc-multilib-mips64-linux-gnuabi64/focal 4:9.2.1-4.1 amd64
  GNU C compiler for the mips64 architecture

gcc-multilib-mips64el-linux-gnuabi64/focal 4:9.2.1-4.1 amd64
  GNU C compiler for the mips64el architecture

gcc-multilib-mipsel-linux-gnu/focal 4:9.2.1-4.1 amd64
  GNU C compiler for the mipsel architecture

gcc-multilib-mipsisa32r6-linux-gnu/focal 4:9.2.1-4.1 amd64
  GNU C compiler for the mipsr6 architecture

gcc-multilib-mipsisa32r6el-linux-gnu/focal 4:9.2.1-4.1 amd64
  GNU C compiler for the mipsr6el architecture

gcc-multilib-mipsisa64r6-linux-gnuabi64/focal 4:9.2.1-4.1 amd64
  GNU C compiler for the mips64r6 architecture

gcc-multilib-mipsisa64r6el-linux-gnuabi64/focal 4:9.2.1-4.1 amd64
  GNU C compiler for the mips64r6el architecture

gcc-multilib-powerpc-linux-gnu/focal 4:9.3.0-1ubuntu2 amd64
  GNU C compiler for the powerpc architecture

gcc-multilib-powerpc64-linux-gnu/focal 4:9.3.0-1ubuntu2 amd64
  GNU C compiler for the ppc64 architecture

gcc-multilib-s390x-linux-gnu/focal 4:9.3.0-1ubuntu2 amd64
  GNU C compiler for the s390x architecture

gcc-multilib-sparc64-linux-gnu/focal 4:9.3.0-1ubuntu2 amd64

```

Reply:
`glibc-devel.i686` is also needed for rockylinux-8, if we remove it, llvm compilation will failed with following error

`/usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found`

Reply:
The package name should be  `glibc-devel` , not `glibc-devel.i686` in `el7/el8`.
We don't need to add the Arch-suffix like `i686, x86_64, aarch64` here.

 **x86:**
```
[root@e1d942388e90 bigtop]# yum search glibc-devel                                                                                                                                                            │···························································
================================================== N/S matched: glibc-devel ==================================================│···························································
glibc-devel.i686 : Object files for development using standard C libraries.                                                   │···························································
glibc-devel.x86_64 : Object files for development using standard C libraries.  
```

 **Arm64:**
```
[root@b10f51b7ee3d workspace]# yum search glibc-devel                                                                                                                                                                                                               │·······················································
============================================================== N/S matched: glibc-devel ===============================================================│·······················································
glibc-devel.aarch64 : Object files for development using standard C libraries.   
```

Reply:
@guyuqi When we are compiling llvm in `x86_64` arch, it will also build `i686` binary file.
![image](https://user-images.githubusercontent.com/31196226/181723825-4f2e21b8-25b0-473c-93ed-54d32cc9224f.png)